pkg/streamingpromql/types/stats.go

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/util/stats/query_stats.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package types
+
+// QueryStats tracks statistics about the execution of a single query.
+//
+// It is not safe to use this type from multiple goroutines simultaneously.
+type QueryStats struct {
```

Comment:
Why create our own struct instead of using stats.QuerySamples and passing that to the appropriate operators?

It would also make it easier to implement `TotalSamplesPerStep` if we want to support that too (which I think we do)